blockprod/src/detail/builder.rs

```diff
@@ -214,3 +214,203 @@ impl Drop for BlockMakersDestroyer {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::tests::setup_blockprod_test;
+    use chainstate::BlockSource;
+    use crypto::random::{make_pseudo_rng, Rng};
+    use std::thread;
+    use tokio::task::yield_now;
+    use tokio::time::{timeout, Duration};
+
+    use common::{
+        chain::block::{timestamp::BlockTimestamp, BlockReward, ConsensusData},
+        primitives::{Idable, H256},
+        time_getter::TimeGetter,
+    };
+
+    use super::*;
+
+    #[test]
+    fn test_new_tip() {
+        let (_manager, chain_config, chainstate, mempool) = setup_blockprod_test();
+
+        let (_tx_builder, rx_builder) = mpsc::unbounded_channel();
+
+        let mut builder = PerpetualBlockBuilder::new(
+            chain_config,
+            chainstate,
+            mempool,
+            Default::default(),
+            rx_builder,
+            true,
+        );
+
+        let block_id = Id::new(H256::from_low_u64_be(make_pseudo_rng().gen()));
```

Comment:
you can use `H256::random_using`

Reply:
That's a nice convenience function, thanks will update.